sway/commands/bar/colors.c

```diff
@@ -16,38 +18,53 @@ static struct cmd_handler bar_colors_handlers[] = {
 	{ "urgent_workspace", bar_colors_cmd_urgent_workspace },
 };
 
+static char *hex_to_rgba_hex(const char *hex) {
+	uint32_t color;
+	if (!parse_color(hex, &color)) {
+		return NULL;
+	}
+	char *rgba = malloc(10);
```

Comment:
No `NULL` check?

Reply:
Send a PR to add one?

Reply:
Sure, I can do. Do `ENOMEM` errors have to be propagated via `cmd_results`? If so, it seems calling `cmd_results_new()` would fail with `ENOMEM` too. Is there a way to signal failure without allocating?

Reply:
Hmm, that's an issue. Could we return a NULL result to indicate an allocation failure?

Reply:
I guess so. It seems this pattern is just being copied from elsewhere in the codebase though.

For example:

https://github.com/swaywm/sway/blob/31a83bd48d9aad2a039565fc39f1a52bd0cdf17d/sway/criteria.c#L50-L53

https://github.com/swaywm/sway/blob/31a83bd48d9aad2a039565fc39f1a52bd0cdf17d/sway/criteria.c#L504-L507

A quick search with `git grep '(re|c|m)alloc *\(' -- sway/` shows up dozens more.

Reply:
Yes, NULL checks are missing throughout the whole codebase.